content/knowledge/flare_architecure.md

```diff
@@ -0,0 +1,207 @@
+* No cooperation needed from an underlying chain's validators: Validators from an underlying chain do not need to modify their chain's codebase to permit Flare to interpret their network.
+  An underlying chain's validators do not even need to be aware that Flare exists for the state connector system to operate.
+* Can read the state of any blockchain: The state connector can operate on any possible Sybil-resistance technique of an underlying chain.
+  For example proof-of-work, proof-of-stake, and even federated byzantine agreement where there is no global agreement on the set of validators in control of a network.
+* No encoding of the current validators in control of an underlying chain to a smart contract on Flare: This requirement of other state-relay approaches such as the SPV proof leads to the hazardous scenario where the enforcement of bad behavior in relaying state needs to be conducted by the same set of operators that have performed the bad behavior.
+* Constant-sized proofs: both the data availability proof and the payment proof are constant-sized, independent of the number of other payments in the data availability period being considered.
+* Every Flare validator independently verifies an underlying chain's state: If your Flare validator observes the canonical state of an underlying chain, then you will not lose safety against that chain.
+
+### Design
+
+State Connector is built into the validators.
+These validators (or validation nodes) are tasked with ordering transactions and confirming validity for consensus plus digesting state changes from integrated blockchains.
+Validators on the Flare Network operate on the ecosystem as well as the integrated blockchains’ ecosystems.
+
+The validation process consists of two stages:
+
+* congestion control by agreement on data availability
+* proving a transaction
+
+#### Congestion Control by Agreement on Data Availability
+
+In this stage, validators agree on the state of the underlying chain.
+This process of tracking state change availability occurs via a smart contract on the Ethereum Virtual Machine (EVM) level.
+The integrated blockchain validators can submit a valid claim availability proof in a competition to be first but only allows them to do so again after the timeout period has ended since the last valid claim availability proof.
+Additionally, a fee to submit the availability proofs is required to prevent attacking and spamming.
+The timeout period, fee, and count of finalized blocks from the integrated blockchains can all be changed via governance.
+
+#### Proving a Transaction
+
+After enough validators agree on the state of the network (data availability proof), validators need to approve the transaction.
+Validators will agree/disagree on the validity of the transaction using the Federated Byzantine Agreement for the network to reach a consensus.
```

Comment:
```suggestion
Validators use the Federated Byzantine Agreement to reach a consensus on whether the transaction is valid.
```

Reply:
More details of the `merkle trie` would be great as I am missing the context here. Is it for the underlying blockchain?